test/services/DateServiceSpec.scala

```diff
@@ -38,35 +43,126 @@ class DateServiceSpec extends TestSupport with FeatureSwitching {
-    "return the next year date if the timeMachineAddYears 1" in {
-      //enable(TimeMachineAddYear)
-      val f = fixture(fixedDate.toString)
-//      f.mockedTestDateService.getCurrentDate  should equal(fixedDate.plusYears(1))
-      f.mockedTestDateService.getCurrentDate  should equal(fixedDate.plusYears(0))
+  "The getCurrentDate method when TimeMachine FS is off" should {
+    "return the current date and time" in {
+      setupGetCurrentTestMocks()
+
+      val getCurrentDate = TestDateService.getCurrentDate
+
+      getCurrentDate shouldBe LocalDate.now()
     }
-    "return the mocked current date" in {
-      val f = fixture("2018-03-29")
-      //enable(TimeMachineAddYear)
-//      f.mockedTestDateService.getCurrentDate should equal(LocalDate.parse("2019-03-29"))
-      f.mockedTestDateService.getCurrentDate should equal(LocalDate.parse("2018-03-29"))
+  }
+
+  "The getCurrentDate method when TimeMachine FS is on" should {
+    "return the current date and time" when {
+      "the TimeMachineAddYear and TimeMachineSetDate FS are off" in {
+        setupGetCurrentTestMocks()
+
+        val getCurrentDate = TestDateService.getCurrentDate
+
+        getCurrentDate shouldBe LocalDate.now()
+      }
+    }
+    "return the current date plus a year" when {
+      "the TimeMachine is on, and we have set the TimeMachine to add an additional year" in {
+        setupGetCurrentTestMocks(timeMachineFS = true, timeMachineAddYearFS = true, addYears = 2)
+
+        val expectedDate = LocalDate.now().plusYears(2)
+
+        TestDateService.getCurrentDate shouldBe expectedDate
+      }
+    }
+    "return a LocalDate of the current year, but on the 3rd of April" when {
+      "the TimeMachine is on, and we have set the TimeMachine to place us on the 3rd of April" in {
+        setupGetCurrentTestMocks(timeMachineFS = true, timeMachineSetDateFS = true, setDay = 3, setMonth = 4)
+
+        val expectedDate = LocalDate.of(LocalDate.now().getYear, 4, 3)
+
+        TestDateService.getCurrentDate shouldBe expectedDate
+      }
+    }
+    "return a LocalDate of the current year, but on the 27th of August" when {
+      "the TimeMachine is on, and we have set the TimeMachine to place us on the 27th of August" in {
+        setupGetCurrentTestMocks(timeMachineFS = true, timeMachineSetDateFS = true, setDay = 27, setMonth = 8)
+
+        val expectedDate = LocalDate.of(LocalDate.now().getYear, 8, 27)
+
+        TestDateService.getCurrentDate shouldBe expectedDate
+      }
+    }
+  }
+  "return a LocalDate, two calendar years in the future time on the 9th of April" when {
+    "the TimeMachine is on, and we have set the TimeMachine to place us on a specific date with additional years" in {
+      setupGetCurrentTestMocks(timeMachineFS = true, timeMachineAddYearFS = true, addYears = 2, timeMachineSetDateFS = true, setDay = 9, setMonth = 4)
+
+      val expectedDate = LocalDate.of(LocalDate.now.plusYears(2).getYear, 4, 9)
```

Comment:
+ try added/implement next method to the DateServiceInterface
++ call it as a single source of truth while evaluating date, you can override it where needed in the UTs
```
trait DateServiceInterface {
  protected def now(): LocalDate
```

Reply:
You said it was OK to use this if it won't break in the future, which these won't. For testing the other methods in the class (aka ones that aren't getCurrentDate) then I would not use this approach

Reply:
I would really suggest to avoid use of dynamic data/generated data, again, this is not fit in the way we use UT or ITs;
its is expected to use static input, its hard to understand how it works and support this code in the futurel 
Dynamic data / generated input is fit under category of property based testing, and this is not property base test;

Reply:
How am I supposed to unit test this then? I can't mock a response from LocalDate.now() so I don't know how I can test/property based test

Reply:
you don't have to mock LocalDate.now(), you can apply change proposed above, like:
and override => protected def now(): LocalDate in the fake you are going to use in the tests
```
trait DateServiceInterface {
  protected def now(): LocalDate

  def getCurrentDate: LocalDate = now()
```

Reply:
Ah ok I'll give it a go